### DISCUSSION TITLE
When using JsonProperty attribute, SelectMany does not use the property name when querying

### DISCUSSION THREAD

**Discussion by cacodev** [JasperFx/marten#1740] [Q&A]
If I have a model..

```csharp
public class MyAppModel
{
    [JsonProperty("id")]
    public Guid Id { get; set; }
    [JsonProperty("name")]
    public string Name { get; set; }
    [JsonProperty("childObjects")]
    public IList<ChildObject> Children { get; set; } = new List<ChildObject>();
}
```

When this model is stored. The json is serialized in the data column as expected. Reading the data, works as expected.

However, when I try to use `.SelectMany()` to select just the child objects, no results are returned.

If I use the `.ToCommand()` function I see that the sql generated looks like this:

```sql
select jsonb_array_elements_text(CAST(d.data ->> 'Children' as jsonb)) as x from public.mt_doc_myappmodel as d
```

seems that the property selector ('Children') should use the property name from the attribute instead ('childObjects')

**Comment by jeremydmiller** [comment]
@cacodev Marten doesn't use `[JsonProperty]` whatsoever in formulating Linq queries. I'd advise you not to do that with persistent documents.
